plonky2/src/recursion/dummy_circuit.rs

```diff
@@ -0,0 +1,69 @@
+use alloc::vec;
+
+use plonky2_field::extension::Extendable;
+use plonky2_util::ceil_div_usize;
+
+use crate::gates::noop::NoopGate;
+use crate::hash::hash_types::RichField;
+use crate::iop::witness::{PartialWitness, Witness};
+use crate::plonk::circuit_builder::CircuitBuilder;
+use crate::plonk::circuit_data::{CircuitData, CommonCircuitData};
+use crate::plonk::config::GenericConfig;
+use crate::plonk::proof::ProofWithPublicInputs;
+
+/// Generate a proof for a dummy circuit. The `public_inputs` parameter let the caller specify
+/// certain public inputs (identified by their indices) which should be given specific values.
+/// The rest will default to zero.
+pub(crate) fn dummy_proof<F, C, const D: usize>(
+    circuit: &CircuitData<F, C, D>,
+    nonzero_public_inputs: &[(usize, F)],
```

Comment:
Is there a reason to not use a HashMap here? To avoid the quadratic search below.

Reply:
I was thinking these should be pretty small, but yeah maybe best to use `HashMap` just in case